### PR TITLE
Shims a package useful for asserting pkg params work

### PR DESCRIPTION
**Pull request by steviecoaster**
## Usage

* `choco pack pipeline.nuspec`
* `choco install pipeline -y --package-parameters="'/File=foo.txt'"`
* `choco uninstall pipeline -y --package-parameters="'/File=foo.txt'"`

This should install a file called foo.txt to $env:TEMP, and similarly remove it

Adjust a `--source` parameter as necesary.

### CODE REVIEW

**Comment by steviecoaster** [comment]
Supplements fix for #8

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 /tests/output
 .vagrant/
 .testresults/
+*.nupkg
```
Comment (vexx32) [inline .gitignore:5]: Files must end in a newline, please. 🙂

**Comment by vexx32** [comment]
@steviecoaster might be better to put this in a subfolder here: https://github.com/chocolatey/chocolatey-ansible/tree/master/chocolatey/tests/integration/targets/win_chocolatey/files

Note there's already a package there, so we could figure something out in terms of moving that into a subdirectory or something and adjusting the paths on whichever test targets that package... _or_ we could just merge this package with the existing one, so it can be used for both tests.

**Comment by vexx32** [comment]
This is erroring on install, it would seem:
```
Access to the path 'C:\\Users\\ansible-test\\AppData\\Local\\Temp\\chocolatey' is denied."
```

Full error:

```
"Running 'ChocolateyScriptRunner' for ansible v0.1.0 with packageScript 'C:\\ProgramData\\chocolatey\\lib\\ansible\\tools\\chocolateyinstall.ps1', packageFolder:'C:\\ProgramData\\chocolatey\\lib\\ansible', installArguments: '', packageParameters: '',",
        "Running 'C:\\ProgramData\\chocolatey\\lib\\ansible\\tools\\chocolateyinstall.ps1'",
        "Running Get-PackageParameters ",
        "Parsing $env:ChocolateyPackageParameters and $env:ChocolateyPackageParametersSensitive for parameters",
        "Running Get-PackageParameters ",
        "Parsing $env:ChocolateyPackageParameters and $env:ChocolateyPackageParametersSensitive for parameters",
        "ERROR: The running command stopped because the preference variable \"ErrorActionPreference\" or common parameter is set to Stop: Access to the path 'C:\\Users\\ansible-test\\AppData\\Local\\Temp\\chocolatey' is denied.",
        " at <ScriptBlock>, C:\\ProgramData\\chocolatey\\lib\\ansible\\tools\\chocolateyinstall.ps1: line 59",
        "at <ScriptBlock>, C:\\ProgramData\\chocolatey\\helpers\\chocolateyScriptRunner.ps1: line 49",
        "at <ScriptBlock>, <No file>: line 1",

```

File: chocolatey/tests/integration/targets/win_chocolatey/files/tools/chocolateyinstall.ps1 [hunk 2]
```diff
@@ -53,3 +54,7 @@ $package_info = @{
 $package_json = ConvertTo-Json -InputObject $package_info
 
 [System.IO.File]::WriteAllText($install_path, $package_json)
+
+if($pp){
+    New-Item -Path $env:TEMP -Name $($pp['File'])
+}
```
Comment (vexx32) [inline chocolatey/tests/integration/targets/win_chocolatey/files/tools/chocolateyinstall.ps1:60]:
```suggestion
$params = $package_info['package_params']
if($params['File']){
    New-Item -Path $env:TEMP -Name $package_info['package_params']['File']
}
```

File: chocolatey/tests/integration/targets/win_chocolatey/files/tools/chocolateyinstall.ps1 [hunk 3]
```diff
@@ -1,5 +1,6 @@
 $ErrorActionPreference = 'Stop'
 
+$pp = Get-PackageParameters
```
Comment (vexx32) [inline chocolatey/tests/integration/targets/win_chocolatey/files/tools/chocolateyinstall.ps1:3]:
```suggestion
```

File: chocolatey/tests/integration/targets/win_chocolatey/files/tools/chocolateyUninstall.ps1 [hunk 4]
```diff
@@ -1,9 +1,17 @@
 $ErrorActionPreference = 'Stop'
 
+$pp = Get-PackageParameters
```
Comment (vexx32) [inline chocolatey/tests/integration/targets/win_chocolatey/files/tools/chocolateyUninstall.ps1:3]:
```suggestion
$params = Get-PackageParameters
```

File: chocolatey/tests/integration/targets/win_chocolatey/files/tools/chocolateyUninstall.ps1 [hunk 5]
```diff
@@ -1,9 +1,17 @@
...
 
 if (Test-Path -LiteralPath $install_path) {
     Remove-Item -LiteralPath $install_path -Force > $null
 }
+
+if($pp){
+    if (Test-Path -LiteralPath "$env:TEMP\$($pp['File'])") {
+        Remove-Item -LiteralPath "$env:TEMP\$($pp['File'])" -Force > $null
+    }
+}
```
Comment (vexx32) [inline chocolatey/tests/integration/targets/win_chocolatey/files/tools/chocolateyUninstall.ps1:17]:
```suggestion
if ($params['File']) {
    if (Test-Path -LiteralPath "$env:TEMP\$($params['File'])") {
        Remove-Item -LiteralPath "$env:TEMP\$($params['File'])" -Force > $null
    }
}
```

**Comment by vexx32** [comment]
Having had a chance to look at the tests we have currently, I think this PR is mostly not needed right now. The test package already writes out package parameters during its install, and we have test cases ensuring that the package parameters are being passed along correctly.

Thanks for submitting this, but closing this for now as it looks like it mostly is not necessary to include here. 🙂

**Pull request closed without merging**